Runtime/JSValueScope.cs

```diff
@@ -1,50 +1,103 @@
 using System;
+using System.Threading;
 using static NodeApi.JSNativeApi.Interop;
 
 namespace NodeApi;
 
-public class JSValueScope : IDisposable
+public enum JSValueScopeType { Handle, Escapable, Callback, Root, RootNoContext, }
+
+public sealed class JSValueScope : IDisposable
```

Comment:
Can `JSValueScope` be a struct also (now that there is no longer a class hierarchy)? That would save a heap allocation on every callback.

Reply:
I would like it to be. It cannot be a `struct` because we reference it from `JSValue`. I want to explore if we can make it `ref struct`, but I want to do it in a follow up PR. This one already grown too much. :)